catalog/ui/src/app/Catalog/CatalogItemWorkshopForm.tsx

```diff
@@ -0,0 +1,628 @@
+
+  const catalogNamespace:CatalogNamespace = useSelector(
+    (state) => selectCatalogNamespace(state, catalogItem.metadata.namespace)
+  );
+  const userGroups:string[] = useSelector(selectUserGroups);
+  const userIsAdmin:boolean = useSelector(selectUserIsAdmin);
+  const userRoles:string[] = useSelector(selectUserRoles);
+  const workshopNamespaces:ServiceNamespace[] = useSelector(selectWorkshopNamespaces);
+
+  const submitRequestEnabled:boolean = checkEnableSubmit(formState);
+
+  async function submitRequest(): Promise<void> {
+    if (!submitRequestEnabled) {
+      throw "submitRequest called when submission should be disabled!";
+    }
+    const parameterValues:any = {};
+    for (const parameterState of Object.values(formState.parameters)) {
+      // Add parameters for request that have values and are not disabled or hidden
+      if (parameterState.value !== undefined && !parameterState.isDisabled && !parameterState.isHidden && !(parameterState.value === '' && !parameterState.isRequired)) {
+        parameterValues[parameterState.name] = parameterState.value;
+      }
+    }
+
+    const workshop:Workshop = await createWorkshop({
+      accessPassword: workshopAccessPassword,
+      catalogItem: catalogItem,
+      description: workshopDescription,
+      displayName: workshopDisplayName,
+      openRegistration: userRegistrationValue === 'open',
+      // FIXME - Allow selecting service namespace
+      serviceNamespace: workshopNamespaces[0],
+    });
+
+    const workshopProvision:WorkshopProvision = await createWorkshopProvision({
+      catalogItem: catalogItem,
+      concurrency: workshopProvisionConcurrency,
+      count: workshopProvisionCount,
+      parameters: parameterValues,
+      startDelay: workshopProvisionStartDelay,
+      workshop: workshop,
+    });
+
+    history.push(`/workshops/${workshop.metadata.namespace}/${workshop.metadata.name}`);
+  }
+
+  async function checkConditions(): Promise<void> {
+    try {
+      await checkConditionsInFormState(formState, userGroups, userIsAdmin, userRoles);
+      dispatchFormState({
+        type: "checkConditionsComplete",
+      });
+    } catch (error) {
+      setErrorMessage(`Failed evaluating condition in form ${error}`);
+    }
+  }
+
+  // First render and detect unmount
+  useEffect(() => {
+    return () => {
+      componentWillUnmount.current = true;
+    }
+  }, []);
+
+  // Initialize form groups for parameters and default vaules
+  React.useEffect(() => {
+    setErrorMessage(undefined);
+    dispatchFormState({
+      type: "init",
+      catalogItem: catalogItem,
+    });
+  }, [catalogItem.metadata.uid]);
+
+  React.useEffect(() => {
+    if (formState) {
+      if (!formState.conditionChecks.complete) {
+        checkConditions();
+      }
+      return () => {
+        if (componentWillUnmount.current) {
+          cancelFormStateConditionChecks(formState);
+        }
+      }
+    } else {
+      return null;
+    }
+  }, [formState]);
```

Comment:
im not sure this will detect object changes as expected, will be better to send directly the [formState.conditionChecks.complete] or use a deep compare strategy (maybe if the object is shallow enough convert the object to string)

Reply:
The `formState` is a new object each time because of how the reducer works. I am not happy with this code, but it does work as expected. I think it could be done better. Also there is duplication here and the `CatalogItemRequestForm.tsx` that should be cleaned up, but I think that is work for a later PR.

Reply:
Makes sense, is ok like this then 👌🏼